.circleci/config.yaml

```diff
@@ -0,0 +1,120 @@
+version: 2.1
+commands:
+  install-cloudfriend:
+    description: "Install mapbox/cloudfriend"
+    steps:
+      - run: npm install -g @mapbox/cfn-config
+jobs:
+  build:
+    working_directory: /usr/src/app
+    docker:
+      - image: python:3-stretch
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Install dependencies
+          command: |
+              # Install latest LTS node
+              apt-get update
+              curl -sL https://deb.nodesource.com/setup_10.x | bash -
+              apt-get install -y nodejs libgeos-dev # Required for shapely
+              node --version
+      - restore_cache:
+          keys:
+            - cached-directories
+          paths:
+            - $CIRCLE_WORKING_DIRECTORY/client/node_modules
+            - $CIRCLE_WORKING_DIRECTORY/env
+      - run:
+          name: Install requirements
+          command: |
+            # Install NPM packages and build client from gulpfile
+            cd $CIRCLE_WORKING_DIRECTORY/client
+            npm install
+            ./node_modules/.bin/gulp build
+            cd $CIRCLE_WORKING_DIRECTORY
+            # Install Python dependencies
+            pip install virtualenv
+            virtualenv env
+            $CIRCLE_WORKING_DIRECTORY/env/bin/pip install --upgrade pip
+            $CIRCLE_WORKING_DIRECTORY/env/bin/pip install -r requirements.txt
+      - run:
+          name: Run tests
+          command: |
+            # JS Unit Tests
+            cd $CIRCLE_WORKING_DIRECTORY/tests/client
+            ../../client/node_modules/.bin/karma start ./karma.conf.js \
+              --single-run --browsers PhantomJS --reporters junit
+            # Run Python tests
+            cd $CIRCLE_WORKING_DIRECTORY
+            env/bin/nosetests ./tests/server --with-xunit \
+              --xunit-file $CIRCLE_TEST_REPORTS/unitresults.xml \
+              --with-coverage --cover-erase --cover-package=./server
+            env/bin/coverage xml -o $CIRCLE_TEST_REPORTS/coverage.xml
+      - save_cache:
+          key: cached-directories
+          paths:
+            - $CIRCLE_WORKING_DIRECTORY/client/node_modules
+            - $CIRCLE_WORKING_DIRECTORY/env
+  deploy-stage:
+    working_directory: /usr/src/app
+    docker:
+      - image: python:3-stretch
+    steps:
+      - checkout
+      - setup_remote_docker
+      - install-cloudfriend
+      - run: 
+          name: Remove last snapshot and backup database
+          command: |
+            # Delete old snapshot
+            aws rds delete-db-snapshot tm3-staging-deployment-snapshot-latest
+            # create new aws rds snapshot
+            aws rds create-db-snapshot --db-snapshot-identifier tm3-staging-deployment-snapshot-latest --db-instance-identifier tasking-manager-staging 
```

Comment:
Should we create a snapshot, before deleting the old one? This would mean some difficulties with naming, but would be safe in the scenario that the snapshot deletion was successful but the snapshot creation was not.

Reply:
This can be fixed by copying the old one to a temp name, then doing the deployment, then safely deleting the temp old one.

Reply:
Awesome. 👍